### PR TITLE
Feature/INESRP-113_Improve_GCov_plugin_overall_coverage_validation_2

### PR DESCRIPTION
**Pull request by danielmartinezruiz**
Related to https://github.com/zaleos/Ceedling/pull/7 using a stable branch.

### CODE REVIEW

File: plugins/gcov/gcov.rake [hunk 1]
```diff
@@ -179,29 +179,57 @@ namespace UTILS_SYM do
...
+      fail_under_line = 0
+    end
+    fail_under_branch = @ceedling[:configurator].project_config_hash[:gcov_fail_under_branch] || 0
+    if fail_under_branch < 0 || fail_under_branch > 100
+      puts "Minimum coverage by branch (#{fail_under_branch}%) has to be a percentage from 0 to 100."
+      fail_under_branch = 0
+    end
+
+    extra_params = []
+    if fail_under_line + fail_under_branch > 0
```
Comment (tobyjwebb) [inline plugins/gcov/gcov.rake]: Readability issue?
I would use:
if fail_under_line > 0 or fail_under_branch > 0

**Review by tobyjwebb — CHANGES_REQUESTED** [review]

**Comment by tobyjwebb** [comment]
Oh, wait... No unit tests?

File: plugins/gcov/README.md [hunk 2]
```diff
@@ -58,6 +58,17 @@ coverage, use the following config.
 
 These HTML and XML reports will be found in `build/artifacts/gcov`.
 
+It is possible to allow gcov to check the coverage percentage of the source
+files and exit Ceedling with an error. Use the following config; both are
```
[the remark this answers is not in the document]
Comment (danielmartinezruiz) [reply plugins/gcov/README.md]: Done the first one.

The second one I think it is not needed, cause all Ceedling (and coverture) are for CI processes.

File: plugins/gcov/gcov.rake (hunk 1 above)
Comment (danielmartinezruiz) [reply plugins/gcov/gcov.rake]: Refactored.

**Review by tobyjwebb — APPROVED** [review]

**Comment by danielmartinezruiz** [comment]
As someone did this open-source work on the upstream, it has no sense to keep this branch/PR

**Pull request closed without merging**